lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -325,36 +325,36 @@ class OperationsPolicyElementAssociation < ::ActiveRecord::Base
         self.buffering? ? klass.create_later(element_attrs, self) : klass.create(element_attrs)
       end
 
-
       define_method("find_all_of_type_#{pe_type}") do |options = {}|
         conditions = options.slice!(:per_page, :page, :ignore_case).stringify_keys
         extra_attribute_conditions = conditions.slice!(*PolicyElement.column_names)
         include_conditions, conditions = conditions.partition { |k,v| include_condition?(k,v) }
         pe_class = class_for_type(pe_type)
 
         # Arel matches provides agnostic case insensitive sql for mysql and postgres
-        all = begin
-          if options[:ignore_case]
-            match_expressions = conditions.map {|k,v| ignore_case_applies?(options[:ignore_case],k) ?
-              pe_class.arel_table[k].matches(v) : pe_class.arel_table[k].eq(v) }
-            match_expressions.inject(pe_class.where(nil)) {|rel, e| rel.where(e)}
-          else
-            pe_class.where(conditions.to_h)
-          end
-        end
+        all = if options[:ignore_case]
+                conditions.map do |k,v|
+                  if ignore_case_applies?(options[:ignore_case], k)
+                    pe_class.arel_table[k].matches(v)
+                  else
+                    pe_class.arel_table[k].eq(v)
+                  end
+                end.inject(pe_class.where(nil)) { |rel, e| rel.where(e) }
+              else
+                pe_class.where(conditions.to_h)
+              end
 
         include_conditions.each do |key, value|
           all = Adapter.apply_include_condition(scope: all, key: key, value: value[:include], klass: class_for_type(pe_type))
         end
 
         extra_attribute_conditions.each do |key, value|
-          Warn.once("WARNING: #{self.class} is filtering #{pe_type} on #{key} in memory, which won't scale well. " <<
-            "To move this query to the database, add a '#{key}' column to the policy_elements table " <<
-            "and re-save existing records")
-            all.to_a.select!{ |pe| pe.store_attributes and
-                        ((attr_value = pe.extra_attributes_hash[key]).is_a?(String) and
-                        value.is_a?(String) and ignore_case_applies?(options[:ignore_case],key)) ? attr_value.downcase == value.downcase : attr_value == value}
+          Warn.once("WARNING: #{self.class} is filtering #{pe_type} on #{key} in memory, which won't scale well. " \
+                    "To move this query to the database, add a '#{key}' column to the policy_elements table " \
+                    "and re-save existing records")
+          all.to_a.select! { |pe| pe_matches_extra_attributes?(pe, key, value, options[:ignore_case]) }
```

Comment:
I pulled out this logic chain into a method because it was so dense. Not necessary for the pluck, but necessary for the eyes.